File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    int n;
    vector<vector<int>> res;
    void dfs(int k, int sum, vector<int>& nums, int target, vector<int>& temp){
        if(sum==target){
            res.push_back(temp);
            return;
        }
        if(sum>target|| k==n)return;
        int num = nums[k];
        dfs(k+1,sum,nums,target,temp);
        for(int i = 1;i<= (int)((double)target/(double)num);++i){
            sum+=num;
            temp.push_back(num);
            dfs(k+1,sum,nums,target,temp);
        }
        for(int i=1;i<=(int)((double)target/(double)num);++i){
            temp.pop_back();
            sum-=num;
        }
        
    }
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        n = nums.size();
        vector<int> temp;
        vector<bool> visited(n,false);
        dfs(0,0,nums,target,temp);
        return res;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers) 
        L,R = 0,n-1
        temp = numbers[L]+numbers[R]
        while (temp != target):
            if temp>target:
                R -=1
            if temp < target:
                L+=1
            temp = numbers[L]+numbers[R]
        return [L+1,R+1]
        